content/tutorial-x-ray-image-processing.md

```diff
@@ -0,0 +1,571 @@
+   should probably use `np.stack()`, as evidenced when you try to time the code
+   with Python's `timeit`:
+
+    ```python
+    %timeit combined_xray_images_1 = np.stack([file1, file2, file3, file4, file5, file6, file7, file8])
+    ```
+
+    Example output:
+
+    ```
+    1.52 ms ± 49.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
+    ```
+
+    ```python
+    %timeit C = [combined_xray_images_2.append(imageio.imread(os.path.join(DIR, '00000011_00'+str(i)+'.png'))) for i in range(8)]
+    ```
+
+    Example output:
+
+    ```
+    159 ms ± 2.69 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
+    ```
+
++++
+
+**2.** Check the shape of the new X-ray image array containing 8 stacked images:
+
+```{code-cell} ipython3
+combined_xray_images_1.shape
+```
+
+**3.** You can now display the "health progress" by plotting each of frames next
+to each other using Matplotlib:
+
+```{code-cell} ipython3
+fig, axes = plt.subplots(nrows=1, ncols=8, figsize=(30, 30))
+
+for i in range(8):
+    x = combined_xray_images_1[i]
+    axes[i].imshow(x, cmap='gray')
+    axes[i].axis('off')
+```
+
+**4.** In addition, it can be helpful to show the progress as an animation.
+Let's create a GIF file with `imageio.mimwrite()` and display the result in the
+notebook:
+
+```{code-cell} ipython3
+from IPython.display import Image
+
+GIF_PATH = os.path.join(DIR, 'xray_image.gif')
+imageio.mimwrite(GIF_PATH, combined_xray_images_1, format= '.gif', fps = 1) 
+
+Image(filename=GIF_PATH, width=400, height=400)
+```
+
+## Edge detection using the Laplacian-Gaussian, Gaussian gradient, Sobel, and Canny filters
+
++++
+
+When processing biomedical data, it can be useful to emphasize the 2D
+["edges"](https://en.wikipedia.org/wiki/Edge_detection) to focus on particular
+features in an image. To do that, using
+[image gradients](https://en.wikipedia.org/wiki/Image_gradient) can be
+particularly helpful when detecting the change of color pixel intensity.
+
++++
+
+### The Laplace filter with Gaussian second derivatives
+
+Let's start with an n-dimensional
+[Laplace](https://en.wikipedia.org/wiki/Laplace_distribution) filter
+("Laplacian-Gaussian") that uses
+[Gaussian](https://en.wikipedia.org/wiki/Normal_distribution) second
+derivatives. This Laplacian method focuses on pixels with rapid intensity change
+in values and is combined with Gaussian smoothing to
+[remove noise](https://homepages.inf.ed.ac.uk/rbf/HIPR2/log.htm). Let's examine
+how it can be useful in analyzing 2D X-ray images.
+
++++
+
+- The implementation of the Laplacian-Gaussian filter is relatively
+straightforward: 1) import the `ndimage` module from SciPy; and 2) call
+[`scipy.ndimage.gaussian_laplace()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.gaussian_laplace.html)
+with a sigma (scalar) parameter, which affects the standard deviations of the
+Gaussian filter (you'll use `1` in the example below):
```

Comment:
Also note that the leading `- ` [turns this paragraph into a bullet](https://59-248354526-gh.circle-artifacts.com/0/site/_build/html/content/tutorial-x-ray-image-processing.html#the-laplace-filter-with-gaussian-second-derivatives) - not sure if that's intentional or not!

Reply:
The dash is intentional. 

The steps were numbered to help users follow the train of thought, since the sentence wasn't a simple one. It does look better now after the suggestion 👍 

Thanks !